testdata/go1.23/main.go

```diff
@@ -8,6 +10,15 @@ func testFuncRange(f func(yield func(int) bool)) {
 	for i := range f {
 		println(i)
 	}
+	next, stop := iter.Pull(f)
+	defer stop()
+	for {
+		i, ok := next()
+		if !ok {
+			break
+		}
+		println(i)
+	}
```

Comment:
Can you put this in a separate function? `testFuncRange` is for testing range over func, you could add a new `testCoro` for example.

Reply:
Done.